seed/serialize.py

```diff
@@ -44,6 +45,73 @@ def validate(seed):
             print("platform not in ", PLATFORMS)
             return False
 
+    feature_names_to_studies = collections.defaultdict(list)
+    for study in seed['studies']:
+        used_feature_names = set()
+        for experiment in study['experiments']:
+            feature_association = experiment.get('feature_association')
+            if feature_association:
+                for enable_feature in feature_association.get('enable_feature', []):
+                    used_feature_names.add(enable_feature)
+                for disable_feature in feature_association.get('disable_feature', []):
+                    used_feature_names.add(disable_feature)
+
+        for used_feature_names in used_feature_names:
+            feature_names_to_studies[used_feature_names].append(study)
+
+    def get_study_platforms(study):
+        return set(study.get('filter', {}).get('platform', []))
+
+    def get_study_channels(study):
+        return set(study.get('filter', {}).get('channel', []))
+
+    def get_study_version_range(study):
+        def version_to_int_array(version_str):
+            parts = version_str.split('.')
+            version_list = []
+            for part in parts:
+                if part == '*':
+                    break
```

Comment:
Hm, I'm not 100% sure about this part.
`*` means different for min/max version.
as for Python is alway `[1,2] > [1]`

Reply:
> Hm, I'm not 100% sure about this part. `*` means different for min/max version. as for Python is alway `[1,2] > [1]`

yes, you're right. This comparison is tricky. Fixed and add tests from the C++ version.